File: simple-stack-example-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/injection/AndroidModule.java
package com.zhuinden.simplestackdemoexamplefragments.application.injection;

import android.content.Context;
import android.content.res.Resources;

import com.zhuinden.simplestackdemoexamplefragments.application.CustomApplication;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Owner on 2017. 01. 27..
 */

@Module
public class AndroidModule {
    @Provides
    @Named("applicationContext")
    public Context applicationContext() {
        return CustomApplication.get();
    }

    @Provides
    public Resources resources(@Named("applicationContext") Context context) {
        return context.getResources();
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/injection/AndroidModule.kt
package com.zhuinden.simplestackdemoexamplefragments.application.injection

import android.content.Context
import android.content.res.Resources
import com.zhuinden.simplestackdemoexamplefragments.application.CustomApplication
import dagger.Module
import dagger.Provides
import javax.inject.Named

/**
 * Created by Owner on 2017. 01. 27..
 */
@Module
object AndroidModule {
    @Provides
    @Named("applicationContext")
    @JvmStatic
    fun applicationContext(): Context = CustomApplication.get()

    @Provides
    @JvmStatic
    fun resources(@Named("applicationContext") context: Context): Resources = context.resources
}
